### PR TITLE
Recordings re-sync: Migrated the re-sync recordigs to `ServerRoomRow`.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Testing Steps

> 1. Pull the code.
> 2. Install the dependencies `bundle install && npm|yarn install`.
> 3. Clean the previous assets build by running `rm app/assets/builds/*` (This won't remove .keep since it's hidden).
> 4. Clean the database and tmp files for a better isolation by running `rails tmp:clear && rails db:schema:cache:clear && rails db:drop && rails db:create && rails db:migrate:with_data`
> 5. Run the linter and specs `bundle exec rubocop --parallel && bundle exec rspec && npx eslint app/javascript/* --ext .jsx,.js`
> 6. Run `./bin/dev` to run the assets builders processes and the Puma server all at once.

## Screenshots (if appropriate):

image

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3957

File: app/javascript/hooks/mutations/admin/server_recordings/useRecordingsReSync.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import toast from 'react-hot-toast';
+import { useMutation, useQueryClient } from 'react-query';
+
+import axios from '../../../../helpers/Axios';
+
+export default function useRecordingsReSync(friendlyId) {
+  const queryClient = useQueryClient();
+
+  return useMutation(
+    () => axios.get(`/admin/server_rooms/${friendlyId}/resync.json`),
```
Comment (KH-Amir-TN) [inline app/javascript/hooks/mutations/admin/server_recordings/useRecordingsReSync.jsx:11]: This is a great example of an API that mutates our server(s) state and vulnerable to CSRF attacks even with our mitigation,

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #3969

**Review by scouillard — APPROVED** [review]

File: app/javascript/hooks/mutations/admin/server_recordings/useRecordingsReSync.jsx (hunk 1 above)
Comment (farhatahmad) [reply app/javascript/hooks/mutations/admin/server_recordings/useRecordingsReSync.jsx:11]: Agreed

**Review by farhatahmad — APPROVED** [review]

**Pull request merged**